### PR TITLE
Add mergent registration

### PR DESCRIPTION
**Pull request by PascalThesing**

### CODE REVIEW

File: src/Merchants/Content/Merchant/Services/RegistrationService.php [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+    public function registerMerchant(RequestDataBag $requestDataBag, SalesChannelContext $salesChannelContext): string {
+        $requestDataWithDefaultValues = $this->createDefaultDataForMerchant($requestDataBag);
+
+        $customerId = $this->customerRegistrationService->register($requestDataWithDefaultValues, false, $salesChannelContext);
+
+        $parameters = $requestDataBag->all();
+        $parameters['id'] = Uuid::randomHex();
+        $parameters['customerId'] = $customerId;
+        $parameters['salesChannelId'] = $salesChannelContext->getSalesChannel()->getId();
```
Comment (JanPietrzyk) [inline src/Merchants/Content/Merchant/Services/RegistrationService.php]: the sales channel needs to be selected by the merchant and submitted through the api. Only the customer mus be scoped in the default (API) sales channel

File: src/Merchants/Migration/Migration1585212510DoubleOptIn.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+class Migration1585212510DoubleOptIn extends MigrationStep
+{
+    public function getCreationTimestamp(): int
+    {
+        return 1585212510;
+    }
+
+    public function update(Connection $connection): void
+    {
+        $connection->insert('system_config', [
```
Comment (shyim) [inline src/Merchants/Migration/Migration1585212510DoubleOptIn.php]: I guess we cannot do this anymore. Cause we will have also customers now. So we have to enable that only fly, when a merchant will be registered

File: src/Merchants/Content/Merchant/Services/RegistrationService.php [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+        $this->entityRepository->create([$parameters], $salesChannelContext->getContext());
+
+        return $parameters['id'];
+    }
+
+    private function validateMerchantData(array $merchantData): void {
```
Comment (shyim) [inline src/Merchants/Content/Merchant/Services/RegistrationService.php]: Can you also add the validation part?

**Comment by PascalThesing** [comment]
@shyim why should is?

**Pull request merged**